collector/adcs.go

```diff
@@ -0,0 +1,246 @@
+}
+
+func (c *ADCSCollector) collectADCSCounters(ctx *ScrapeContext, ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
+	dst := make([]perflibADCS, 0)
+	if _, ok := ctx.perfObjects["Certification Authority"]; !ok {
+		return nil, errors.New("Perflib did not contain an entry for Certification Authority")
+	}
+	err := unmarshalObject(ctx.perfObjects["Certification Authority"], &dst)
+	if err != nil {
+		return nil, err
+	}
+	if len(dst) == 0 {
+		return nil, errors.New("Perflib query for Certification Authority (ADCS) returned empty result set")
+	}
+
+	for _, d := range dst {
+		n := strings.ToLower(d.Name)
+		if n == "" {
+			continue
+		}
+		ch <- prometheus.MustNewConstMetric(
+			c.RequestsPerSecond,
+			prometheus.CounterValue,
+			d.RequestsPerSecond,
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.RequestProcessingTime,
+			prometheus.GaugeValue,
+			milliSecToSecADCS(d.RequestProcessingTime),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.RetrievalsPerSecond,
+			prometheus.CounterValue,
+			d.RetrievalsPerSecond,
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.RetrievalProcessingTime,
+			prometheus.GaugeValue,
+			milliSecToSecADCS(d.RetrievalProcessingTime),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.FailedRequestsPerSecond,
+			prometheus.CounterValue,
+			d.FailedRequestsPerSecond,
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.IssuedRequestsPerSecond,
+			prometheus.CounterValue,
+			d.IssuedRequestsPerSecond,
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.PendingRequestsPerSecond,
+			prometheus.CounterValue,
+			d.PendingRequestsPerSecond,
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.RequestCryptographicSigningTime,
+			prometheus.GaugeValue,
+			milliSecToSecADCS(d.RequestCryptographicSigningTime),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.RequestPolicyModuleProcessingTime,
+			prometheus.GaugeValue,
+			milliSecToSecADCS(d.RequestPolicyModuleProcessingTime),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.ChallengeResponsesPerSecond,
+			prometheus.CounterValue,
+			d.ChallengeResponsesPerSecond,
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.ChallengeResponseProcessingTime,
+			prometheus.GaugeValue,
+			milliSecToSecADCS(d.ChallengeResponseProcessingTime),
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.SignedCertificateTimestampListsPerSecond,
+			prometheus.CounterValue,
+			d.SignedCertificateTimestampListsPerSecond,
+			d.Name,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.SignedCertificateTimestampListProcessingTime,
+			prometheus.GaugeValue,
+			milliSecToSecADCS(d.SignedCertificateTimestampListProcessingTime),
+			d.Name,
+		)
+	}
+
+	return nil, nil
+}
+
+func milliSecToSecADCS(t float64) float64 {
+	return t / 1000
+}
```

Comment:
There's a duplicate `milliSecToSec` function in the `remote_fx` exporter.

We could move it to the `collector/collector.go` file and re-use to avoid duplication.

Reply:
Moved to common `collector/collector.go`